k8s-plugin/src/main/kotlin/com/amazon/spinnaker/keel/k8s/model/K8sResourceModel.kt

```diff
@@ -17,13 +17,15 @@ data class K8sResourceModel(
 )
 
 typealias K8sSpec = MutableMap<String, Any?>
+typealias K8sData = MutableMap<String, Any>
 
 data class K8sObjectManifest(
     val apiVersion: String,
     val kind: String,
     @get:ExcludedFromDiff
     val metadata: Map<String, Any?>,
-    val spec: K8sSpec
+    val spec: K8sSpec?,
+    val data: K8sData? = null
```

Comment:
do we need the duplication for `K8sData` and `K8sSpec` here?

why not a more unified pattern like:

```
kind: k8s/credential@v1
spec:
  metadata:
    application: "moretest"
  template:
    metadata:
      namespace: keel-dev
    spec:
      account: manabu-git-repo
```
I know it is more verbose, but is consistent with other k8s resources we have. Also, I noticed, the name for the k8s secret is driven from the account name. account names in clouddriver are only unique per provider. so if we want docker accounts and git accounts, there will be name conflicts. does it make sense to also have a type introduced for the secret, similar to what we have for artifacts? or can we do something else to dedup secrets?

Reply:
Good point. We should definitely have type defined to deal with duplicate names. 
`K8sObjectManifest` is used as the manifest to be submitted to kubernetes cluster. Secrets and configmaps do not have spec field so do we not need the data field? `K8sData` and `K8sSpec` are there in case we need to make specific changes to one but not the other.

Reply:
how about we jam data under template like the following:

```
kind: k8s/credential@v1
spec:
  metadata:
    application: "moretest"
  template:
    metadata:
      namespace: keel-dev
      type: git
    data:
      account: manabu-git-repo
```

Reply:
I am inclined to go with `spec.template.metadata.type`. I think it makes sense to have it closer together with other relevant fields for better clarity

Reply:
ok this looks good. or we can make `apiVersion` and `kind` be nullable if that is more concise. your call.